
introduce -Wundef check to catch macro typos
more infos about motivation at https://sourceware.org/glibc/wiki/Wundef

patch 03/04: use #ifdef for autoconf provided marcos
until autoconf switch from un/define marco logic to always existing macro with 0/1 value

Signed-off-by: Andreas Mohr <and@gmx.li>

--- a/lib/unixcompat.h
+++ b/lib/unixcompat.h
@@ -13,9 +13,9 @@
 
 #include <sys/types.h>          /* BSD */
 
-#if MAJOR_IN_MKDEV
+#ifdef MAJOR_IN_MKDEV
 #include <sys/mkdev.h>
-#elif MAJOR_IN_SYSMACROS
+#elif defined MAJOR_IN_SYSMACROS
 #include <sys/sysmacros.h>
 #endif
 
--- a/src/filemanager/filegui.c
+++ b/src/filemanager/filegui.c
@@ -53,9 +53,9 @@
 #include <config.h>
 
 /* Keep this conditional in sync with the similar conditional in m4.include/mc-get-fs-info. */
-#if ((STAT_STATVFS || STAT_STATVFS64)                                       \
-     && (HAVE_STRUCT_STATVFS_F_BASETYPE || HAVE_STRUCT_STATVFS_F_FSTYPENAME \
-         || (! HAVE_STRUCT_STATFS_F_FSTYPENAME)))
+#if ((defined STAT_STATVFS || defined STAT_STATVFS64)                                       \
+     && (defined HAVE_STRUCT_STATVFS_F_BASETYPE || defined HAVE_STRUCT_STATVFS_F_FSTYPENAME \
+         || (! defined HAVE_STRUCT_STATFS_F_FSTYPENAME)))
 #define USE_STATVFS 1
 #else
 #define USE_STATVFS 0
@@ -85,7 +85,7 @@
 #include <nfs/nfs_clnt.h>
 #include <nfs/vfs.h>
 #endif
-#elif HAVE_OS_H                 /* BeOS */
+#elif defined HAVE_OS_H                 /* BeOS */
 #include <fs_info.h>
 #endif
 
@@ -134,12 +134,12 @@
 #endif
 #endif
 
-#if HAVE_STRUCT_STATVFS_F_BASETYPE
+#ifdef HAVE_STRUCT_STATVFS_F_BASETYPE
 #define STATXFS_FILE_SYSTEM_TYPE_MEMBER_NAME f_basetype
 #else
-#if HAVE_STRUCT_STATVFS_F_FSTYPENAME || HAVE_STRUCT_STATFS_F_FSTYPENAME
+#if defined HAVE_STRUCT_STATVFS_F_FSTYPENAME || defined HAVE_STRUCT_STATFS_F_FSTYPENAME
 #define STATXFS_FILE_SYSTEM_TYPE_MEMBER_NAME f_fstypename
-#elif HAVE_OS_H                 /* BeOS */
+#elif defined HAVE_OS_H                 /* BeOS */
 #define STATXFS_FILE_SYSTEM_TYPE_MEMBER_NAME fsh_name
 #endif
 #endif
diff -ruN a/src/subshell.c b/src/subshell.c
--- a/src/subshell.c
+++ b/src/subshell.c
@@ -601,10 +601,10 @@
 
 #ifdef HAVE_POSIX_OPENPT
     pty_master = posix_openpt (O_RDWR);
-#elif HAVE_GETPT
+#elif defined HAVE_GETPT
     /* getpt () is a GNU extension (glibc 2.1.x) */
     pty_master = getpt ();
-#elif IS_AIX
+#elif defined IS_AIX
     strcpy (pty_name, "/dev/ptc");
     pty_master = open (pty_name, O_RDWR);
 #else
